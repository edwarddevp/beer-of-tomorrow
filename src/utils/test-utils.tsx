import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "../theme";
import { createSerializer, matchers } from "@emotion/jest";
import { useStore } from "@/configs/store";
import { Provider } from "react-redux";
import mockBeerItems from "./mock-beer-items";
// @ts-ignore
expect.addSnapshotSerializer(createSerializer());
// @ts-ignore
expect.extend(matchers);

const initialStoreValues = {
  getBeerById: {
    data: mockBeerItems[0],
    isLoading: false,
    errors: [],
  },
};

const AllTheProviders: FC = ({ children }) => {
  const store = useStore(initialStoreValues);
  return (
    <ChakraProvider theme={theme}>
      <Provider store={store}>{children}</Provider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
