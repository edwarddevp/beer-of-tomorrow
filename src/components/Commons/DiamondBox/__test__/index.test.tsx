import React from "react";
// Using render and screen from test-utils.js instead of
// @testing-library/react
import { render } from "test-utils";
import { DiamondBox } from "../";

describe("DiamondBox", () => {
  it("renders", () => {
    const { container } = render(<DiamondBox />);
    expect(container).toMatchSnapshot();
  });
  it("renders gray bg", () => {
    const { container } = render(<DiamondBox bg="gray.500" />);
    expect(container).toMatchSnapshot();
  });
  it("renders size 8", () => {
    const { container } = render(<DiamondBox size={8} />);
    expect(container).toMatchSnapshot();
  });
});
