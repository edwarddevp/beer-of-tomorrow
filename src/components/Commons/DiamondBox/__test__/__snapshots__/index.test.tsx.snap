// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiamondBox renders 1`] = `
.emotion-0 {
  width: var(--chakra-sizes-2);
  height: var(--chakra-sizes-2);
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  background: var(--chakra-colors-primary-500);
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`DiamondBox renders gray bg 1`] = `
.emotion-0 {
  width: var(--chakra-sizes-2);
  height: var(--chakra-sizes-2);
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  background: var(--chakra-colors-gray-500);
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`DiamondBox renders size 8 1`] = `
.emotion-0 {
  width: var(--chakra-sizes-8);
  height: var(--chakra-sizes-8);
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  background: var(--chakra-colors-primary-500);
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
